### ISSUE TITLE
训练semidet----dense_teacher 损失出现nan值

### ISSUE THREAD

**Issue by zranwang** [PaddlePaddle/PaddleDetection#7942]
### 问题确认 Search before asking

- [X] 我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleDetection/issues)，没有发现相似的bug。I have searched the [issues](https://github.com/PaddlePaddle/PaddleDetection/issues) and found no similar bug report.

### Bug组件 Bug Component

Training

### Bug描述 Describe the Bug

[03/14 15:03:10] ppdet.engine INFO: Epoch: [3] [300/726] learning_rate: 0.010000 loss: 1.490179 loss_sup_sum: 1.490179 loss_unsup_sum: 0.000000 fg_sum: 0.000000 loss_cls: 0.492379 loss_box: 0.385261 loss_quality: 0.626925 distill_loss_cls: 0.000000 distill_loss_box: 0.000000 distill_loss_quality: 0.000000 eta: 14:11:18 batch_cost: 0.2928 data_cost: 0.0032 ips: 6.8317 images/s
[03/14 15:03:25] ppdet.engine INFO: Epoch: [3] [350/726] learning_rate: 0.010000 loss: 1.449128 loss_sup_sum: 1.449128 loss_unsup_sum: 0.000000 fg_sum: 0.000000 loss_cls: 0.453509 loss_box: 0.362126 loss_quality: 0.626894 distill_loss_cls: 0.000000 distill_loss_box: 0.000000 distill_loss_quality: 0.000000 eta: 14:10:07 batch_cost: 0.2809 data_cost: 0.0036 ips: 7.1203 images/s
[03/14 15:03:40] ppdet.engine INFO: Epoch: [3] [400/726] learning_rate: 0.010000 loss: nan loss_sup_sum: nan loss_unsup_sum: 0.000000 fg_sum: 0.000000 loss_cls: nan loss_box: 0.960481 loss_quality: nan distill_loss_cls: 0.000000 distill_loss_box: 0.000000 distill_loss_quality: 0.000000 eta: 14:09:36 batch_cost: 0.2921 data_cost: 0.0032 ips: 6.8481 images/s
[03/14 15:03:55] ppdet.engine INFO: Epoch: [3] [450/726] learning_rate: 0.010000 loss: nan loss_sup_sum: nan loss_unsup_sum: 0.000000 fg_sum: 0.000000 loss_cls: nan loss_box: 0.973398 loss_quality: nan distill_loss_cls: 0.000000 distill_loss_box: 0.000000 distill_loss_quality: 0.000000 eta: 14:08:56 batch_cost: 0.2895 data_cost: 0.0033 ips: 6.9095 images/s
[03/14 15:04:10] ppdet.engine INFO: Epoch: [3] [500/726] learning_rate: 0.010000 loss: nan loss_sup_sum: nan loss_unsup_sum: 0.000000 fg_sum: 0.000000 loss_cls: nan loss_box: 0.972789 loss_quality: nan distill_loss_cls: 0.000000 distill_loss_box: 0.000000 distill_loss_quality: 0.000000 eta: 14:08:39 batch_cost: 0.2959 data_cost: 0.0031 ips: 6.7587 images/s
[03/14 15:04:25] ppdet.engine INFO: Epoch: [3] [550/726] learning_rate: 0.010000 loss: nan loss_sup_sum: nan loss_unsup_sum: 0.000000 fg_sum: 0.000000 loss_cls: nan loss_box: 0.972375 loss_quality: nan distill_loss_cls: 0.000000 distill_loss_box: 0.000000 distill_loss_quality: 0.000000 eta: 14:07:56 batch_cost: 0.2881 data_cost: 0.0035 ips: 6.9422 images/s
[03/14 15:04:41] ppdet.engine INFO: Epoch: [3] [600/726] learning_rate: 0.010000 loss: nan loss_sup_sum: nan loss_unsup_sum: 0.000000 fg_sum: 0.000000 loss_cls: nan loss_box: 0.972025 loss_quality: nan distill_loss_cls: 0.000000 distill_loss_box: 0.000000 distill_loss_quality: 0.000000 eta: 14:08:05 batch_cost: 0.3042 data_cost: 0.0037 ips: 6.5748 images/s

### 复现环境 Environment

-os:linux
-paddlepaddle:2.3.2
-paddleDetection: release/2.6
-pathon: 3.7

### Bug描述确认 Bug description confirmation

- [X] 我确认已经提供了Bug复现步骤、代码改动说明、以及环境信息，确认问题是可以复现的。I confirm that the bug replication steps, code change instructions, and environment information have been provided, and the problem can be reproduced.

### 是否愿意提交PR？ Are you willing to submit a PR?

- [ ] 我愿意提交PR！I'd like to help by submitting a PR!

**Comment by zranwang** [comment]
没有更改过代码，每次固定在 Epoch: [3] [400/726] 出现nan值

**Comment by nemonameless** [comment]
训练命令也请发下，是训coco数据集？

**Comment by zranwang** [comment]
训练指令：python -m paddle.distributed.launch --log_dir=denseteacher_fcos_semi010/ --gpus 0,1,2,3,4,5,6,7 tools/train.py -c configs/semi_det/denseteacher/denseteacher_fcos_r50_fpn_coco_semi010.yml --eval

训练coco数据集

**Comment by nemonameless** [comment]
torch原版我们也训过偶尔也会出现nan。可以这样改再去试试，https://github.com/PaddlePaddle/PaddleDetection/blob/develop/configs/semi_det/denseteacher/denseteacher_fcos_r50_fpn_coco_semi010.yml#L11
```
### pretrain and warmup config, choose one and comment another
pretrain_weights: https://bj.bcebos.com/v1/paddledet/models/fcos_r50_fpn_2x_coco_sup010.pdparams # mAP=26.3
semi_start_iters: 0
ema_start_iters: 0
use_warmup: &use_warmup False

# pretrain_weights: https://paddledet.bj.bcebos.com/models/pretrained/ResNet50_cos_pretrained.pdparams
# semi_start_iters: 5000
# ema_start_iters: 3000
# use_warmup: &use_warmup True
```

**Comment by zranwang** [comment]
好的，谢谢你，我试试

**Comment by zranwang** [comment]
同样出现nan值，试了两次

[03/14 19:38:06] ppdet.engine INFO: Epoch: [3] [350/726] learning_rate: 0.010000 loss: 2.352705 loss_sup_sum: 1.236267 loss_unsup_sum: 1.119772 fg_sum: 99.939156 loss_cls: 0.375279 loss_box: 0.293459 loss_quality: 0.559709 distill_loss_cls: 0.353088 distill_loss_box: 0.216404 distill_loss_quality: 0.549518 eta: 23:33:41 batch_cost: 0.4795 data_cost: 0.0048 ips: 4.1710 images/s
Error: /paddle/paddle/phi/kernels/gpu/bce_loss_kernel.cu:42 Assertion `(x >= static_cast<T>(0)) && (x <= one)` failed. Input is expected to be within the interval [0, 1], but recieved nan.

**Comment by nemonameless** [comment]
加载了26.3权重预训练，还是会卡在 Epoch: [3] [350/726] 吗？paddle版本也升级下

**Comment by zranwang** [comment]
加载了26.3权重预训练，还是会卡在 Epoch: [3] [350/726] ，paddle版本也升级到2.4.2也是同样卡在 Epoch: [3] [350/726]

**Comment by wjm202** [comment]
在以下环境下直接使用26.3权重预训练，测试后是可以运行的，不会出nan，如果还是有问题的话可以尝试进入configs/semi_det/denseteacher/denseteacher_fcos_r50_fpn_coco_semi010.yml 降低loss_weight 和学习率大小
运行环境
cuda 10.2 
paddlepaddle-gpu    2.3.2
运行指令
python -m paddle.distributed.launch --log_dir=denseteacher_fcos_semi010/ --gpus 0,1,2,3,4,5,6,7 tools/train.py -c configs/semi_det/denseteacher/denseteacher_fcos_r50_fpn_coco_semi010.yml --eval

**Comment by zranwang** [comment]
按照你说那样试了，还有一样的问题，即卡在 Epoch: [3] [350/726] 。相同代码和数据，学习率改成0.00125，做了以下两个尝试：1.单机单卡就可以正常训练；2.改成多机多卡（8张卡）同样卡在Epoch: [3] [350/726]

**Comment by lyf6** [comment]
在自己的数据集上做测试dense teacher掉点，请问有遇到过这种情况的吗
